lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php

```diff
@@ -53,6 +53,14 @@ public function connect(array $params, $username = null, $password = null, array
                 $pdo->setAttribute(PDO::PGSQL_ATTR_DISABLE_PREPARES, true);
             }
 
+            /* defining client_encoding via SET NAMES to avoid inconsistent DSN support
+             * - the 'client_encoding' connection param only works with postgres >= 9.1
+             * - passing client_encoding via the 'options' param breaks pgbouncer support
+             */
+            if (isset($params['charset'])) {
+              $pdo->query('SET NAMES \''.$params['charset'].'\'');
```

Comment:
@riccardonar remember -- the reason this all came up is because pgbouncer does not support the 'options' param. The options param is out of the question. We either need to use client_encoding, or SET NAMES.

I am all for building out support for both. I'll move forward with the fallback feature (on rejection of client_encoding) and reconnect AFTER you guys confirm PGSQL_LIBPQ_VERSION is not feasible (as it reflects the libpq version of the CONNECTING server, not the database server).